plugins/services/src/js/pages/EditFrameworkConfiguration.js

```diff
@@ -0,0 +1,143 @@
+import React, { PropTypes } from "react";
+import mixin from "reactjs-mixin";
+import { StoreMixin } from "mesosphere-shared-reactjs";
+import { routerShape } from "react-router";
+import deepEqual from "deep-equal";
+import CosmosPackagesStore from "#SRC/js/stores/CosmosPackagesStore";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import FrameworkConfiguration from "#SRC/js/components/FrameworkConfiguration";
+import Loader from "#SRC/js/components/Loader";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+
+class EditFrameworkConfiguration extends mixin(StoreMixin) {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      packageDetails: null,
+      deployErrors: null,
+      formData: null,
+      formErrors: {},
+      hasError: false
+    };
+
+    this.store_listeners = [
+      {
+        name: "cosmosPackages",
+        events: [
+          "serviceDescriptionError",
+          "serviceDescriptionSuccess",
+          "serviceUpdateSuccess",
+          "serviceUpdateError"
+        ]
+      }
+    ];
+
+    CosmosPackagesStore.fetchServiceDescription(props.params.id);
+  }
+
+  onCosmosPackagesStoreServiceDescriptionSuccess() {
+    const fullPackage = CosmosPackagesStore.getServiceDetails();
+    const packageDetails = new UniversePackage(fullPackage.package);
+    const formData = this.reorderResolvedOptions(
+      fullPackage.resolvedOptions,
+      packageDetails.getConfig()
+    );
+
+    this.setState({ packageDetails, formData });
+  }
+
+  onCosmosPackagesStoreServiceDescriptionError() {
+    this.setState({ hasError: true });
+  }
+
+  onCosmosPackagesStoreServiceUpdateSuccess() {
+    this.handleGoBack();
+  }
+
+  onCosmosPackagesStoreServiceUpdateError(deployErrors) {
+    this.setState({ deployErrors });
+  }
+
+  // necessary to have review screen same order of tabs as the form
+  reorderResolvedOptions(resolvedOptions, config) {
```

Comment:
🤔 Out of curiosity, why are these `resolvedOptions` in a different order?Would it help to use a different data structure?

Reply:
I agree and wish this wasn't necessary because it's quite weird to see in code. But when iterating over the keys in the parsed JSON response from Cosmos, the order is different.

We use the `.package` key (the schema) to render the form and tabs. And the `.resolvedOptions` key gives the current running framework configuration. So this method is resetting the resolvedOptions keys to match the order of the schema (the resolvedOptions is used to render the review screen because it's the source of truth of data values).